acestep/ui/gradio/events/results/generation_info.py

```
@@ -23,8 +23,20 @@
+    """Return pre-generation output updates without remounting audio players.
+
+    In Gradio runtime, keep the 8 audio components unchanged via ``gr.skip()``
+    and only clear the batch-download file list (9th output). This avoids
+    replacing player DOM nodes while generation is in progress, which can
+    reset browser-side volume state.
+
+    In non-Gradio test environments, gracefully fall back to 9 ``None`` values.
+    """
+    try:
+        import gradio as gr  # local import keeps headless tests dependency-free
+    except Exception:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)